src/app/components/ClientConfigLoader.tsx

```diff
@@ -4,8 +4,12 @@ import { ClientConfig } from '../hooks/useClientConfig';
 import { trimTrailingSlash } from '../utils/common';
 
 const getClientConfig = async (): Promise<ClientConfig> => {
-  const url = `${trimTrailingSlash(import.meta.env.BASE_URL)}/config.json`;
-  const config = await fetch(url, { method: 'GET' });
+  let url = `${trimTrailingSlash(import.meta.env.BASE_URL)}/config.${window.location.hostname}.json`;
+  let config = await fetch(url, { method: 'GET' });
+  if(config.status===404) {
```

Comment:
It's worth noting that in the dev environment, `npm start` stupidly returns a `200` for this non-existent file. This isn't something that would happen in a production environment (I'd certainly hope at least!), but it does cause the dev server to throw an error while loading.

Might it be worth reading the JSON before returning it (or trying to) to see if the returned config is ACTUALLY 200 OK?

Reply:
it doesn't look like Element does this, so if I'm sticking to the original goal of basically just mirroring their behaviour, then this is sufficient.